cpp/src/arrow/util/align_util_test.cc

```diff
@@ -59,6 +63,18 @@ void CheckBitmapWordAlign(const uint8_t* data, int64_t bit_offset, int64_t lengt
   }
 }
 
+arrow::Result<std::shared_ptr<Array>> CreateUnalignedArray(const Array& array) {
+  BufferVector sliced_buffers(array.data()->buffers.size(), nullptr);
```

Comment:
```suggestion
  // Slicing by 1 would create an invalid array if the type was wider than
  // 1 byte so double-check that the array is a 1-byte type
  EXPECT_EQ(array.type_id(), Type::UINT8);
  BufferVector sliced_buffers(array.data()->buffers.size(), nullptr);
```

Reply:
made the change, thanks for the suggestion.